File: library/src/main/java/com/github/windsekirun/rxsociallogin/intenal/utils/Preconditions.java
package com.github.windsekirun.rxsociallogin.intenal.utils;

import android.os.Looper;
import android.support.annotation.RestrictTo;

import io.reactivex.Observer;
import io.reactivex.disposables.Disposables;

import static android.support.annotation.RestrictTo.Scope.LIBRARY_GROUP;

@RestrictTo(LIBRARY_GROUP)
public final class Preconditions {
    public static void checkNotNull(Object value, String message) {
        if (value == null) {
            throw new NullPointerException(message);
        }
    }

    public static boolean checkMainThread(Observer<?> observer) {
        if (Looper.myLooper() != Looper.getMainLooper()) {
            observer.onSubscribe(Disposables.empty());
            observer.onError(new IllegalStateException(
                    "Expected to be called on the main thread but was " + Thread.currentThread().getName()));
            return false;
        }
        return true;
    }

    private Preconditions() {
        throw new AssertionError("No instances.");
    }
}

File: library/src/main/java/com/github/windsekirun/rxsociallogin/intenal/utils/Preconditions.kt
package com.github.windsekirun.rxsociallogin.intenal.utils

import android.os.Looper
import android.support.annotation.RestrictTo

import io.reactivex.Observer
import io.reactivex.disposables.Disposables

import android.support.annotation.RestrictTo.Scope.LIBRARY_GROUP

@RestrictTo(LIBRARY_GROUP)
class Preconditions private constructor() {

    companion object {

        fun <T> checkMainThread(observer: Observer<T>): Boolean {
            if (Looper.myLooper() != Looper.getMainLooper()) {
                observer.onSubscribe(Disposables.empty())
                observer.onError(IllegalStateException(
                        "Expected to be called on the main thread but was " + Thread.currentThread().name))
                return false
            }
            return true
        }
    }
}